docs/src/main/asciidoc/maven-tooling.adoc

```diff
@@ -470,6 +470,36 @@ Uber-Jar creation by default excludes link:https://docs.oracle.com/javase/tutori
 
 Uber-Jar's final name is configurable via a Maven's build settings `finalName` option.
 
+[[multi-module-maven]]
+=== Working with multi-module projects
+
+If your project is a multi-module project, each module that didn't use the Quarkus Maven plugin needs a way to generate a Jandex index to enable CDI bean discovery.
```

Comment:
Are we generating and persisting Jandex index file? There could be projects configured with quarkus-maven-plugin that are also dependencies of another Quarkus app.

Reply:
> Are we generating and persisting Jandex index file?

I don't know how can answer that.

> There could be projects configured with quarkus-maven-plugin that are also dependencies of another Quarkus app.

If I understand it correctly, even dependent project with the `quarkus-maven-plugin` will need the Jandex plugin if we didn't persist the Jandex index. So as a general rules of thumb it will be : _Always include the Jandex plugin in you dependent projet_ ?

Reply:
I don't understand why it always has to be jandex while there is also `META-INF/beans.xml` option.

Reply:
I can recommand the other way. But I want the guide to be straightforward. It's not good to have two ways to do something.
People can still have a look at the CDI guide to find the other way

As I understand it, the plugin is best as it will create the index at build time of the project, where the `META-INF/beans.xml` option will create it at build time of the project having the dependence (so if multiple project depends on it, it will be indexed multiple times).
Also, if a project depends on 10 projects, it will need to build 10 indexes ...

But I can change it to the other way, I just don't want to offers two options because the user will ask which one is the best ... so KISS and document only one in the getting started guide.

Reply:
If we are promoting the Jandex plugin as the way of doing it here then I would probably re-phrase it like:

The best way to enable CDI bean discovery for a module in a multi-module project would be to include the `jandex-maven-plugin`, unless it is the main application module already configured with the `quarkus-maven-plugin`, in which case it will indexed automatically.

Reply:
It misses an introduction sentence, maybe:

> Quarkus is using a *simplified CDI bean discovery*, special treatments needs to be done for Quarkus to discover beans in another module.

or 

> By default, Quarkus will not discover CDI beans inside another module.

Reply:
+1 for thr latter